crates/telio-nurse/src/rtt/ping.rs

```diff
@@ -146,4 +124,74 @@ impl Ping {
 
         dpresults
     }
+
+    async fn ping_action(&self, host: IpAddr) -> PingResults {
+        let mut results = PingResults {
+            successful_pings: 0,
+            unsuccessful_pings: 0,
+            avg_rtt: None,
+        };
+
+        telio_log_debug!("Trying to ping {:?} host", host);
+
+        let mut pinger = match host {
+            IpAddr::V4(_) => {
+                self.client_v4
+                    .clone()
+                    .pinger(host, PingIdentifier(rand::random()))
+                    .await
+            }
+            IpAddr::V6(_) => {
+                self.client_v6
+                    .clone()
+                    .pinger(host, PingIdentifier(rand::random()))
+                    .await
+            }
+        };
+
+        pinger.timeout(Self::PING_TIMEOUT);
+
+        let mut sum = Duration::default();
+
+        for i in 0..self.no_of_tries {
+            match pinger
+                .ping(PingSequence(i.try_into().unwrap_or(0)), &[0; 56])
+                .await
+            {
+                Ok((_, duration)) => {
+                    sum = sum.saturating_add(duration);
+                    results.successful_pings += 1;
+                }
+                Err(e) => {
+                    results.unsuccessful_pings += 1;
+                    telio_log_error!("Ping error: {}", e.to_string());
+                }
+            }
+        }
+
+        results.avg_rtt = sum.checked_div(results.successful_pings);
+        results
+    }
+
+    fn make_builder(proto: ICMP) -> ConfigBuilder {
+        let mut config_builder = PingerConfig::builder().kind(proto);
+        if cfg!(any(
+            target_os = "ios",
+            target_os = "macos",
+            target_os = "tvos",
+        )) {
+            config_builder = config_builder.sock_type_hint(Type::RAW);
```

Comment:
raw socket?

Reply:
Yup, one test was failing when using default (UDP) IPv6 socket on `sock_pool.make_internal()` part.

Reply:
After these changes, it does not error - out, and pings are working.

Reply:
Although, the failing tests were only observed on CI setup (physical M1 MAC worked fine :confused: ).